### PR TITLE
Add multiple selection of skills

### PR DESCRIPTION
**Pull request by rmoldo**
## Description
Added one/multiple selection of skills

## Related
https://trello.com/c/5Oven3Jq

## Link to demo
https://streamable.com/nqdzso

## Steps to Test or Reproduce

1. Open app
2. Tap on skills
3. Select skills

### CODE REVIEW

**Comment by rmoldo** [comment]
[iOS - Skills Filter Screen - Add logic to select skills](https://trello.com/c/5Oven3Jq/108-ios-skills-filter-screen-add-logic-to-select-skills)

File: Shared/Features/Freaks/View/MultipleSelectionRow.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import SwiftUI
+
+struct MultipleSelectionRow: View {
+    @ObservedObject var viewModel = FreaksViewModel()
```
Comment (vladstanescu94) [inline Shared/Features/Freaks/View/MultipleSelectionRow.swift:4]: You should user @ObservedObject only if you pass this class from another view. If you want to instantiate the view model in this view you should use @StateObject.

File: Shared/Features/Freaks/View/MultipleSelectionRow.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    var skill: Skill
+    var isPressed: Bool
```
Comment (vladstanescu94) [inline Shared/Features/Freaks/View/MultipleSelectionRow.swift:6]: I think isSelected sounds better.

File: Shared/Features/Freaks/View/SkillsView.swift [hunk 3]
```diff
@@ -1,26 +1,17 @@
 import SwiftUI
 
 struct SkillsView: View {
-    var viewModel = FreaksViewModel()
+    @ObservedObject var viewModel = FreaksViewModel()
```
Comment (vladstanescu94) [inline Shared/Features/Freaks/View/SkillsView.swift:4]: Same as the other comment about ObservedObject

**Review by vladstanescu94 — CHANGES_REQUESTED** [review]

File: Shared/Features/Freaks/View/MultipleSelectionRow.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import SwiftUI
+
+struct MultipleSelectionRow: View {
+    @StateObject var viewModel = FreaksViewModel()
```
Comment (vladstanescu94) [inline Shared/Features/Freaks/View/MultipleSelectionRow.swift:4]: We are still creating a new instance of the FreaksViewModel inside a child view. I think the better approach is to create the @StateObject in the parent view (Freaks screen) and pass down the reference to the child views as an @ObservedObject

**Review by vladstanescu94 — CHANGES_REQUESTED** [review]

File: Shared/Features/Freaks/View/MultipleSelectionRow.swift (hunk 4 above)
Comment (rmoldo) [reply Shared/Features/Freaks/View/MultipleSelectionRow.swift]: Got it. It should be fine now

**Review by vladstanescu94 — APPROVED** [review]

**Review by DejanuTeodora — APPROVED** [review]

**Review by RalucaMihaela — APPROVED** [review]
Great solution. We should keep in mind that each view should have his ViewModel. We could keep here as it is for now.

**Pull request merged**